### ISSUE TITLE
installation fails on the load_data.php script

### ISSUE THREAD

**Issue by nslocomotives** [sonata-project/sandbox#496]
Please see the output to the console, im afraid i dont yet understand enough about discrimnator mapping to trouble shoot myself any help here would be fab... :-)

Entity 'Sonata\Bundle\DemoBundle\Entity\Car' has to be part of the discriminator map of 'Sonata\Bundle\DemoBundle\Entity\Car' to be properly mapped in the inheritance hierarchy. Alternatively you can make 'Sonata\Bundle\DemoBundle\Entity\Car' an abstract class to avoid this exception from occurring.

**Comment by bambamboole** [comment]
got the same error on 
```
php app/console doctrine:schema:update
```

**Comment by nslocomotives** [comment]
Yea i tired learning about the discriminator stuff, but i haven't got the will power to give it my full attention at the moment, besides id rather spend the time integrating the sonata bundles into my project, i am gutted though i cant get the sandbox running ATM, and would like to see a working example of code on my won machine to poke at.

**Comment by rande** [comment]
@nslocomotives are you using doctrine 2.5 ?

**Comment by rande** [comment]
I just try the sandbox with Doctrine2.5 and I get the same error. You can force doctrine 2.4 while we are fixing this issue.

**Comment by fourpixels** [comment]
I so much hate when I cannot even start the sandbox. I've tried using it like 5 times (through last six months) and I never got it working.

And I hate when NO ONE explains why! Those Sonata guys simply can't understand we don't know all the magic and we can't use it if we don't know how!

I'm so angry, again spent about a day trying to figure this out and then I just dropped it. Total time waster!

**Comment by rande** [comment]
@fourpixels thanks for your comment. The sonata project rely on open source libraries which sometime get updated and break other libs - shit happen. Also, there is no editor behind the Sonata Project so we try to do our best, but sometime we fail.

On this issue, doctrine 2.5 gets release a few days ago, people complains about issues 4 days ago... and you get a solution today.

We also provide a ready to use build https://github.com/sonata-project/sandbox-build/tree/2.4 so you don't have to use composer to get the code.

**Comment by fourpixels** [comment]
Unfortunately I spent 3 hours trying to figure out what's going on, and then realized I copy-pasted the composer command found exactly on that repository, which is wrong:
```php
php composer.phar create-project sonata-project/sandbox:2.4.x-dev
```
This will create project with the regular sandbox, not the pre-build one... Anyways, still having issues with both releases.

Sorry for being rude, but my point is that all of your documentation is totally obsolete! You truly put a lot of efforts on new features and things, but guys - I can't even do a simple admin without looking at StackOverflow for at least 3 things I don't know how to deal with. Your efforts are worthless if people get annoyed trying to install all that bundles with that hundred lines of configuration needed.

Spent few days/weeks on fixing documentation - it would mean much more than new features! Examples are years old. For years there are pages like this: https://sonata-project.org/bundles/admin/2-3/doc/reference/configuration.html saying "THIS DOC PAGE IS NOT UP TO DATE AND WILL BE REMOVED SOON". For years!

Unfortunately there are not much people who can help you. I would love to contribute, but first I must spent weeks or even months browsing all the code and understanding how it works, before I can write any wiki article. And even if I do - 100% I won't have the desire to do it..

Good luck, I hope Sonata will soon become the really easy to use admin, not the one that you must be master with PHP, Symfony and at lest 3 Sonata bundles!

**Comment by rande** [comment]
@fourpixels can you explain where you fail ... at least it will help us to improve more information.  We need execution stack trace, os, some informations ...

**Comment by fourpixels** [comment]
On both versions (sandbox, sandbox-build) I get roughly the same error:

THE ERROR HANDLER HAS CHANGED!
Checking SymlinkScript Mopa\Bundle\BootstrapBundle\Composer\ScriptHandler::postInstallSymlinkTwitterBootstrap handling the post-install-cmd event terminated with an exception

  [Exception]
  D:\work\www\sonata-sandbox\sandbox-build-2.4\vendor/mopa/bootstrap-bundle/Mopa/Bundle/BootstrapBundle\Resources\public\bootstrap exists and is no link!

The regular sandbox version says that `unlink` cannot be executed (on the very same file), because there are no permissions. I'm on windows with admin user, so it's not about permission, don't know what's the issue.

On both cases I delete the file manually and then run `composer install` again, so that post-installation functions can be executed.

Then I get the memory issues while trying to open web/. Raised the limit. Going into admin panel - selecting Categories / Classification:

```php
Fatal error: Call to a member function getContext() on a non-object in D:\work\www\sonata-sandbox\sandbox-build-2.4\vendor\sonata-project\classification-bundle\Controller\CategoryAdminController.php on line 79
Call Stack
#	Time	Memory	Function	Location
1	0.0000	249184	{main}( )	..\app_dev.php:0
2	0.0210	2218032	Symfony\Component\HttpKernel\Kernel->handle( )	..\app_dev.php:29
3	0.4870	11743800	Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel->handle( )	..\bootstrap.php.cache:2377
4	0.4870	11745448	Symfony\Component\HttpKernel\HttpKernel->handle( )	..\bootstrap.php.cache:3133
5	0.4870	11745944	Symfony\Component\HttpKernel\HttpKernel->handleRaw( )	..\bootstrap.php.cache:2984
6	1.3331	41078240	call_user_func_array:{D:\work\www\sonata-sandbox\sandbox-build-2.4\app\bootstrap.php.cache:3022} ( )	..\bootstrap.php.cache:3022
7	1.3331	41079032	Sonata\ClassificationBundle\Controller\CategoryAdminController->treeAction( )	..\bootstrap.php.cache:3022
```

At this point I give up. I've tried using Classification bundle a few months ago, and I couldn't make it - first because it had dependence on Media Bundle, which was not listed, and then because of those context damn ****. There was a 'context' branch, then some other stuff.. Now I don't know. DB table `classification__category` has column `context`, don't know where the problem is.

Sonata logo is missing in the top left, even though I haven't change the path (defaulted when asked during install).

Couldn't dump sample data (`php bin/load_data.php`) because of `xdebug.max_nesting_level is too low` error, even though I am sure I just changed it to 300. This last one may be some other kind of issue - I just got too annoyed to spend more time on debugging.

**Comment by rande** [comment]
@fourpixels thanks, this is the kind of input we want to improve and understand issues.

 - So you should report an issue to the ``MopaBootstrapBundle`` about windows support.
 - The ``CategoryAdminController`` issue is a bug that need to be fixed.
 - The ``xdebug.max_nesting_level`` need to be raised as stated in the official documentation from the symfony's website.

**Comment by nslocomotives** [comment]
@rande yea I am using doctrine 2.5, I have also seen issues with doctrine on other bundles as I'm using php 5.6 on a fedora 21 box, It was to do with the u serialise function, I ended up overriding the code in doctrine to fix it, I'll give 2.4 a try tomorrow if I get a bit of time, I'm enjoying figuring of the admin bundle and how to customise it greatly, it's good work.  If I can get this sandbox running then it wil, help me figure out some more on customising all the modules.  I'm intending on replacing a wordpress / opencart mash up site with a sonata based symfony app, so far it seams to be almost possible.  :-)

**Comment by rande** [comment]
I fix the issue with doctrine 2.5. However the TimelineBundle is not yet compatible with this version. So I force the value to 2.4. But if you remove this bundle you can use doctrine 2.5
